crates/rpc/src/server/mod.rs

```diff
@@ -25,8 +24,8 @@ mod api;
 /// Requests will fail if the components are not available.
 pub struct Rpc {
     pub listener: TcpListener,
-    pub store: SocketAddr,
-    pub block_producer: Option<SocketAddr>,
+    pub store: Url,
+    pub block_producer: Option<Url>,
```

Comment:
nit: Seems like the other components have a `*_url` suffix now, maybe those could be taken out or we could add it to the `Rpc` component as well